src/app/helpers/vcard.js

```diff
@@ -74,37 +74,43 @@ export const getValue = (property) => {
 /**
  * Parse vCard String an return contact properties model as an Array
  * @param {String} vcard to parse
- * @returns {Array} contact properties
+ * @returns {Array} contact properties ordered
  */
 export const parse = (vcard = '') => {
     const comp = new ICAL.Component(ICAL.parse(vcard));
     const properties = comp.getAllProperties();
 
-    return properties.reduce((acc, property) => {
-        const type = property.getParameter('type');
-        const pref = property.getParameter('pref');
-        const splitted = property.name.split('.');
-        const field = splitted[1] ? splitted[1] : splitted[0];
-
-        // Ignore invalid field
-        if (!field) {
-            return acc;
-        }
+    return properties
+        .reduce((acc, property) => {
+            const type = property.getParameter('type');
+            const prefValue = property.getParameter('pref');
+            const pref = typeof prefValue === 'string' ? +prefValue : 0;
+            const splitted = property.name.split('.');
+            const field = splitted[1] ? splitted[1] : splitted[0];
+
+            // Ignore invalid field
+            if (!field) {
+                return acc;
+            }
 
-        const isCustom = isCustomField(field);
+            const isCustom = isCustomField(field);
 
-        // Ignore invalid property
-        if (!isCustom && !PROPERTIES[field]) {
-            return acc;
-        }
+            // Ignore invalid property
+            if (!isCustom && !PROPERTIES[field]) {
+                return acc;
+            }
 
-        const group = splitted[1] ? splitted[0] : undefined;
-        const prop = { pref, field, group, type, value: getValue(property) };
+            const group = splitted[1] ? splitted[0] : undefined;
+            const prop = { pref, field, group, type, value: getValue(property) };
 
-        acc.push(prop);
+            acc.push(prop);
 
-        return acc;
-    }, []);
+            return acc;
+        }, [])
+        .sort((firstEl, secondEl) => {
+            // NOTE Mutation
```

Comment:
why ? you sort the output of a reducer.

Reply:
We need to sort properties by checking pref. I cannot do that in the `reducer` so I do it after. I know I'm doing a mutation, but since `reduce` returns a new Array, it should be fine.